olp-cpp-sdk-core/tests/logging/DisabledLoggingTest.cpp

```diff
@@ -0,0 +1,172 @@
+    OLP_SDK_LOG_INFO("info", "Info "
+                                 << "message");
+    OLP_SDK_LOG_WARNING("warning", "Warning "
+                                       << "message");
+    OLP_SDK_LOG_ERROR("error", "Error "
+                                   << "message");
+
+    // Logs insuppressible by the flag
+    OLP_SDK_LOG_FATAL("fatal", "Fatal "
+                                   << "message");
+    OLP_SDK_LOG_CRITICAL_INFO("info", "Critical info "
+                                          << "message");
+    OLP_SDK_LOG_CRITICAL_WARNING("warning", "Critical warning "
+                                                << "message");
+    OLP_SDK_LOG_CRITICAL_ERROR("error", "Critical error "
+                                            << "message");
+
+    ASSERT_EQ(4U, appender->messages_.size());
+
+    unsigned int index = 0;
+    EXPECT_EQ(Level::Fatal, appender->messages_[index].level_);
+    EXPECT_EQ("fatal", appender->messages_[index].tag_);
+    EXPECT_EQ("Fatal message", appender->messages_[index].message_);
+    EXPECT_NE(std::string::npos, appender->messages_[index].file_.rfind(
+                                     "DisabledLoggingTest.cpp"));
+    EXPECT_LT(0U, appender->messages_[index].line_);
+    EXPECT_NE("DisabledLoggingTest_NoLogging_Test::TestBody()",
+              appender->messages_[index].function_);
+    ++index;
+
+    EXPECT_EQ(Level::Info, appender->messages_[index].level_);
+    EXPECT_EQ("info", appender->messages_[index].tag_);
+    EXPECT_EQ("Critical info message", appender->messages_[index].message_);
+    EXPECT_NE(std::string::npos, appender->messages_[index].file_.rfind(
+                                     "DisabledLoggingTest.cpp"));
+    EXPECT_LT(0U, appender->messages_[index].line_);
+    EXPECT_NE("DisabledLoggingTest_NoLogging_Test::TestBody()",
+              appender->messages_[index].function_);
+    ++index;
+
+    EXPECT_EQ(Level::Warning, appender->messages_[index].level_);
+    EXPECT_EQ("warning", appender->messages_[index].tag_);
+    EXPECT_EQ("Critical warning message", appender->messages_[index].message_);
+    EXPECT_NE(std::string::npos, appender->messages_[index].file_.rfind(
+                                     "DisabledLoggingTest.cpp"));
+    EXPECT_LT(0U, appender->messages_[index].line_);
+    EXPECT_NE("DisabledLoggingTest_NoLogging_Test::TestBody()",
+              appender->messages_[index].function_);
+    ++index;
+
+    EXPECT_EQ(Level::Error, appender->messages_[index].level_);
+    EXPECT_EQ("error", appender->messages_[index].tag_);
+    EXPECT_EQ("Critical error message", appender->messages_[index].message_);
+    EXPECT_NE(std::string::npos, appender->messages_[index].file_.rfind(
+                                     "DisabledLoggingTest.cpp"));
+    EXPECT_LT(0U, appender->messages_[index].line_);
+    EXPECT_NE("DisabledLoggingTest_NoLogging_Test::TestBody()",
+              appender->messages_[index].function_);
+    ++index;
```

Comment:
I dont think you need to test all fields. Only important part here is to test that unsuppressed levels will not reach the appender.
So a simple loop trough the messages with a EXPECT_THAT(level, AnyOf(Eq(Level::Fatal), Eq(Level::Error), ....)) should suffice. Checking that appender messages are correct is part of another test.